File: 205-isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.length() != t.length()){
            return false;
        }
        unordered_map<char,char>MapST;
        unordered_map<char,char>MapTS;
        for(int i=0;i<s.length();i++){
            char c1 = s[i];
            char c2 = t[i];
            // Checking If Mapping Already Exists
            if(MapST.count(c1) && MapST[c1] != c2){
                return false;
            }
            if(MapTS.count(c2) && MapTS[c2] != c1){
                return false;
            }
            MapST[c1]=c2;
            MapTS[c2]=c1;
        }
        return true;
    }
};

File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character,Character>mapST = new HashMap<>();
        HashMap<Character,Character>mapTS = new HashMap<>();
        if(s.length() != t.length()){
            return false;
        }
        for(int i=0;i<s.length();i++){
            char Schar = s.charAt(i);
            char Tchar = t.charAt(i);
            if(mapST.containsKey(Schar)){
                if(mapST.get(Schar) != Tchar){
                    return false;
                }
            }
            else{
                mapST.put(Schar,Tchar);
            }

            if(mapTS.containsKey(Tchar)){
                if(mapTS.get(Tchar) != Schar){
                    return false;
                }
            }
            else{
                mapTS.put(Tchar,Schar);
            }
        }
        return true;
    }
}